qiskit_experiments/framework/composite/parallel_experiment.py

```diff
@@ -97,9 +105,35 @@ def circuits(self):
                     sub_circ = sub_circuits[exp_idx][circ_idx]
                     num_clbits = circuit.num_clbits
                     qubits = sub_qubits[exp_idx]
+                    qargs_map = sub_maps[exp_idx]
                     clbits = list(range(num_clbits, num_clbits + sub_circ.num_clbits))
                     circuit.add_register(ClassicalRegister(sub_circ.num_clbits))
-                    circuit.append(sub_circ, qubits, clbits)
+
+                    # Apply transpiled subcircuit
+                    # Note that this assumes the circuit was not expanded to use
+                    # any qubits outside the specified physical qubits
+                    for inst, qargs, cargs in sub_circ.data:
+                        try:
+                            mapped_qargs = [
+                                circuit.qubits[qargs_map[sub_circ.find_bit(i).index]] for i in qargs
+                            ]
+                        except KeyError as ex:
```

Comment:
This remind me the discussion about coupling map for QV. Sometimes transpiler couples two qubits with swapping via ancillary qubits, this will cause a problem, for example, when `sub_circ` is mapped to the non-adjacent qubits. The simplest solution to this would be automatically generate valid coupling map in the base experiments not to allow such experiment configurations, i.e. ancillary qubits should be explicitly defined in constructor argument `qubits`. If you don't limit the coupling map, please implement the logic and add the test for this edge case.
```python
class EntangleExperiment(BaseExperiment):
    def __init__(qubits, backend):
        super().__init__(qubits, backend=backend):

    def circuits(self):
        qc = QuantumCircuit(2)
        qc.cx(*self.qubits)
        return [qc]

backend = SomeFakeBackend()    # this should be some Falcon 27Q device
exp1 = EntangleExperiment(qubits=[0, 4], backend=backend)
exp2 = EntangleExperiment(qubits=[2, 5], backend=backend)    # if you set 1, 2 instead you'll run into another edge case

pexp = ParallelExperiment([exp1, exp2])
pexp.circuits()   # do some validation for this.
```

Reply:
This is independent of parallel experiment and this sort of validation should happen in base experiment for each individual component experiment.

I just tried testing something towards this but it was much more difficult than expected and ran into a lot of transpiler errors with the coupling pass, so I think this needs to be done later.